### DISCUSSION TITLE
Scroll like pagination

### DISCUSSION THREAD

**Discussion by FrancescoSaverioZuppichini** [qdrant/qdrant#2872] [Q&A] [answered]
Hi there, would it be possible to have something like `scroll` but where I can send a `skip` param instead of the `RecordId` of the last point fetched? This would make pagination way easier

Thanks a lot

**Comment by timvisee** [comment, marked as answer]
No, this is not possible. Instead you should provide the next point ID to search from so Qdrant can continue traversing the list of IDs.

This is a limitation of Qdrant internals. Qdrant goes through all point IDs from beginning to end. Depending on what your data looks like there may be gaps. There is no index of some sort from `0..n` where every index points to a specific point.

For your convenience the API returns `next_page_offset` which you can use as offset for the next request. Please see https://qdrant.tech/documentation/concepts/points/#scroll-points

**Reply by FrancescoSaverioZuppichini** [reply]
Thank you so much

**Comment by msj121** [comment]
I would like to go through my collection and run code using each point.

What if it crashes halfway in 1million vectors? I'd like to store a "page number". 

is "next_page_offset" just a point id and always available if I saved it to hard disk, so I can recover from there, or is next_page_offset a temporary id stored by the database, if so, for how long?

**Reply by msj121** [reply]
Yes - it appears that "next_page_offset" is a specific spot to jump from, like an id or index. At least with my simple tests.

**Reply by timvisee** [reply]
> is "next_page_offset" just a point id and always available if I saved it to hard disk, so I can recover from there, or is next_page_offset a temporary id stored by the database, if so, for how long?

Correct.

Even if that specific point would be deleted it would still resume searching properly.
